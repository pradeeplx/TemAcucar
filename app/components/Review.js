import React from 'react'
import { View, Platform } from 'react-native'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Sentence from "./Sentence"
import TimeAgo from "./TimeAgo"
import ReviewRating from "./ReviewRating"

export default Review = ({ review: { rating, text, reviewer, created_at } }) => (
  <View style={{
    flex: 1,
    flexDirection: 'row',
    alignSelf: 'stretch',
    backgroundColor: Colors.white,
    paddingVertical: 10,
    borderBottomWidth: 12,
    borderColor: Colors.lightBeige,
  }}>
    <View style={{
      flex: 2,
      flexDirection: 'column',
      alignItems: 'center',
    }}>
      <UserImage source={{uri: reviewer.image_url}} />
    </View>
    <View style={{
      flex: 8,
      flexDirection: 'column',
    }}>
      <Sentence style={[{
        fontSize: 12 * fontFactor(),
        color: Colors.blue,
      }, (Platform.OS === 'ios' ? { fontFamily: 'Avenir-Black' } : { fontWeight: 'bold' })]}>
        { `${reviewer.first_name} ${reviewer.last_name}` }
      </Sentence>
      <TimeAgo time={created_at} style={{
        fontSize: 12 * fontFactor(),
        color: Colors.black,
      }} />
      <Sentence style={{
        fontSize: 10 * fontFactor(),
        color: Colors.darkGray,
        marginBottom: 6,
      }}>
        { text || `${reviewer.first_name} não descreveu sua experiência.` }
      </Sentence>
      <ReviewRating rating={rating} />
    </View>
  </View>
)
