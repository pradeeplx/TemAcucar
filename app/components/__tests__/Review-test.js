import 'react-native'
import React from 'react'
import Review from '../Review'
import renderer from 'react-test-renderer'

const review = {
  id: 1,
  rating: 5,
  reviewer: {
    first_name: 'foo',
    last_name: 'bar',
    image_url: '#',
  },
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Review review={review} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders custom text', () => {
  const tree = renderer.create(
    <Review review={{ ...review, text: 'FooBar' }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
