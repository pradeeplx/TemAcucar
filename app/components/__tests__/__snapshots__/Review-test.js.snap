exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderBottomWidth": 12,
      "borderColor": "#FAF6F1",
      "flex": 1,
      "flexDirection": "row",
      "paddingVertical": 10,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "flexDirection": "column",
      }
    }>
    <Image
      source={
        Object {
          "uri": "#",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          undefined,
        ]
      } />
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "flexDirection": "column",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#44B5B1",
              "fontSize": 18,
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      foo bar
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "fontSize": 13.5,
            },
            Object {
              "color": "#000000",
              "fontSize": 18,
            },
          ],
        ]
      }>
      Há segundos
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
            "fontSize": 15,
            "marginBottom": 6,
          },
        ]
      }>
      foo não descreveu sua experiência.
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Ionicons",
              "fontSize": 27,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "color": "#FFBA00",
            },
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Ionicons",
              "fontSize": 27,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "color": "#FFBA00",
            },
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Ionicons",
              "fontSize": 27,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "color": "#FFBA00",
            },
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Ionicons",
              "fontSize": 27,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "color": "#FFBA00",
            },
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Ionicons",
              "fontSize": 27,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "color": "#FFBA00",
            },
          ]
        }>
        
      </Text>
    </View>
  </View>
</View>
`;
